javascript/extraNetworks.js

```diff
@@ -224,16 +224,22 @@ const currentPageForTabs = new Map();
 const totalCountForTabs = new Map();
 let currentTab = 'checkpoints';
 
-async function fetchPageDataAndUpdateList({tabname, model_type, page}) {
+async function fetchPageDataAndUpdateList({tabname, model_type, page, need_refresh = false}) {
     const cardsParentNode = gradioApp().querySelector(`#${tabname}_${model_type}_cards`);
     const searchValue = gradioApp().querySelector('#'+tabname+'_extra_tabs textarea').value.toLowerCase();
     const currentPageTabsId = `${tabname}_${model_type}_current_page`;
     const currentTotalCountId = `${tabname}_${model_type}_total_count`;
     const totalPageNode = gradioApp().querySelector(`#${tabname}_${model_type}_pagination_row .total-page`);
     const currentPageNode = gradioApp().querySelector(`#${tabname}_${model_type}_pagination_row .current-page`);
 
-    const response = await fetch(`/sd_extra_networks/update_page?model_type=${model_type}&page=${page}&search_value=${searchValue}&page_size=${pageSize}`, {
+    const promise = fetch(`/sd_extra_networks/update_page?model_type=${model_type}&page=${page}&search_value=${searchValue}&page_size=${pageSize}&need_refresh=${need_refresh}`, {
         method: "GET", cache: "no-cache"});
+    
+    // loading
+    new AWN().asyncBlock(promise, () => {});

```

Comment:
You don't have to create a new `AWN` object, just use `notifier. asyncBlock()` will be enough. And you can process the response in the callback, like
```
notifier. asyncBlock(promise, (response) => {
    const { model_list, page: resPage, total_count: totalCount, allow_negative_prompt } = await response.json();
});
```